cypress/integration/core-opensearch-dashboards/opensearch-dashboards/left-navigation/left_navigation.spec.js

```diff
@@ -0,0 +1,314 @@
+      cy.contains(/Visualizations/).should('exist');
+
+      // collapse the menu section and expect content to be hidden
+      cy.contains(/Visualize and report/)
+        .should('exist')
+        .click();
+      cy.contains(/Visualizations/).should('not.exist');
+
+      // expand the menu section and expect content to be visible
+      cy.contains(/Visualize and report/)
+        .should('exist')
+        .click();
+      cy.contains(/Visualizations/).should('exist');
+    };
+    if (isWorkspaceEnabled) {
+      createWorkspace('all').then(() => {
+        cy.visit(`w/${workspaceId}/app/all_overview`);
+        validateMenuSection();
+      });
+    } else {
+      cy.visit('app/home');
+      validateMenuSection();
+    }
+  });
+
+  it('navigation should remember state of expand in browser', () => {
+    const validateMenuState = () => {
+      cy.get('.navToggleInLargeScreen').should('exist').click(); // expand the menu
+
+      cy.reload();
+      // navigation menu should remain expanded after reload
+      cy.get('.left-navigation-wrapper').within(() => {
+        isWorkspaceEnabled &&
+          cy.contains(`${workspaceName}_all`).should('exist');
+        cy.get('.navGroupEnabledHomeIcon').should('exist');
+        cy.get('input[type="search"]').should('exist');
+        cy.get('.bottom-container-expanded').should('exist');
+        cy.getElementByTestId('collapsibleNavShrinkButton')
+          .should('exist')
+          .click(); // collapse the menu
+      });
+
+      cy.reload();
+      // navigation menu should remain collapsed after reload
+      cy.get('.left-navigation-wrapper').find('.euiPanel').should('not.exist');
+      cy.get('.left-navigation-wrapper').within(() => {
+        isWorkspaceEnabled &&
+          cy.contains(`${workspaceName}_all`).should('not.exist');
+        cy.get('.navGroupEnabledHomeIcon').should('not.exist');
+        cy.get('input[type="search"]').should('not.exist');
+        cy.get('.bottom-container-expanded').should('not.exist');
+      });
+    };
+
+    if (isWorkspaceEnabled) {
+      createWorkspace('all').then(() => {
+        cy.visit(`w/${workspaceId}/app/all_overview`);
+        validateMenuState();
+      });
+    } else {
+      cy.visit('app/home');
+      validateMenuState();
+    }
+  });
+
+  it('validate navigation history functionality', () => {
+    const validateRecentHistory = () => {
+      // expand the navigation menu
+      cy.get('.navToggleInLargeScreen').should('exist').click();
+      cy.contains(/Visualizations/)
+        .should('exist')
+        .click();
+
+      let visualizationName;
+      cy.getElementByTestId('itemsInMemTable').within(() => {
+        cy.get('.euiLink')
+          .first()
+          .invoke('text')
+          .then((text) => {
+            visualizationName = text;
+            // Click to the first visualization item
+            cy.contains(visualizationName).click();
+          });
+      });
+
+      cy.wait(1000);
```

Comment:
We should use assertion to replace meaningless wait, can we use assertion like visualization title is rendered?

Reply:
Done, this wait seems to be unnecessary, removed